src/fen.ts

```diff
@@ -577,10 +582,130 @@ export const parseFen = (rules: Rules) => (fen: string): Result<Setup, FenError>
   if (rules === 'abalone') {
     return parseAbaloneFen(rules)(fen);
   }
+  if (rules === 'breakthrough') {
+    return parseBreakthroughFen(fen);
+  }
+  if (rules === 'minibreakthrough') {
+    return parseMiniBreakthroughFen(fen);
+  }
+  if (rules === 'linesofaction') {
+    return parseLoAFen(fen);
+  }
+  if (rules === 'scrambledeggs') {
+    return parseScrambledEggsFen(fen);
+  }
 
   return parseDefaultFen(rules)(fen);
 };
 
+const parseBreakthroughFen = (fen: string): Result<Setup, FenError> => {
```

Comment:
all of these 4 functions look almost identical and can be refactored into a single function that takes a couple of extra parameters. Can we do this?